### PR TITLE
GEODE-6568: Use OpenSSL from local install

### PR DESCRIPTION
**Pull request by pivotal-jbarrett**
Support OpenSSL 1.0.1, 1.0.2, 1.1.0, and 1.1.1.

### CODE REVIEW

File: dependencies/CMakeLists.txt [hunk 1]
```diff
@@ -19,7 +19,6 @@ project( dependencies LANGUAGES NONE )
 find_package(Patch REQUIRED)
 
 add_subdirectory(	libxml2 )
```
Comment (pdxcodemonkey) [inline dependencies/CMakeLists.txt]: So we're no longer building OpenSSL in the tree, even if it's not installed on the system?

Comment (pdxcodemonkey) [reply dependencies/CMakeLists.txt]: We'll need to update BUILDING.md to account for installation of OpenSSL, too...

Comment (pivotal-jbarrett) [reply dependencies/CMakeLists.txt]: No, mixing and matching is really hard. Some things in CMake are easy to do in either the configure or build stages but not in both. Yes, need to update the BUILDING.md, good catch!

**Comment by pdxcodemonkey** [comment]
This is filed in Jira as GEODE-6568

**Pull request merged**